compair/models/comparison.py

```diff
@@ -130,12 +152,35 @@ def _get_new_comparison_pair(cls, course_id, assignment_id, user_id, pairing_alg
 
         comparison_pairs = [comparison.convert_to_comparison_pair() for comparison in comparisons]
 
-        comparison_pair = generate_pair(
-            package_name=pairing_algorithm.value,
-            scored_objects=scored_objects,
-            comparison_pairs=comparison_pairs,
-            log=current_app.logger
-        )
+        criterion_scores = {}
+        for answer_with_criterion_score in answers_with_criterion_score:
+            key = answer_with_criterion_score.Answer.id
+            scores = criterion_scores.setdefault(key, {})
+            scores[answer_with_criterion_score.criterion_id] = \
+                answer_with_criterion_score.score
+
+        criterion_weights = {}
+        for the_weight in assignment_criterion_weights:
+            criterion_weights[the_weight.criterion_id] = \
+                the_weight.weight
+
+        # adaptive min delta algo requires extra criterion specific parameters
+        if pairing_algorithm == PairingAlgorithm.adaptive_min_delta:
+            comparison_pair = generate_pair(
```

Comment:
move the `answers_with_criterion_score` and `assignment_criterion_weights` queries along with the `criterion_scores` and  `criterion_weights` in the if so they only run if needed